vllm/engine/async_llm_engine.py

```diff
@@ -37,49 +136,105 @@ class AsyncLLMEngine:
         *args, *kwargs: Arguments for LLMEngine.
     """
 
+    _engine_class: Type[_AsyncLLMEngine] = _AsyncLLMEngine
+
     def __init__(self,
                  worker_use_ray: bool,
                  engine_use_ray: bool,
                  *args,
                  log_requests: bool = True,
+                 inline: bool = False,
                  **kwargs) -> None:
         self.worker_use_ray = worker_use_ray
         self.engine_use_ray = engine_use_ray
         self.log_requests = log_requests
+        self.engine = self._init_engine(*args, **kwargs)
+
+        # Request id -> stream.
+        self.request_streams: Dict[str, AsyncStream] = {}
+        self.finished_requests: Set[str] = set()
+        self.background_loop = None
+        if not inline:
+            # Start the background loop.
+            self.background_loop = asyncio.get_event_loop().create_task(
+                self.run_engine_loop())
+            self.background_loop.add_done_callback(_raise_exception_on_finish)
+
+    def _init_engine(self, *args, **kwargs) -> LLMEngine:
```

Comment:
```suggestion
    def _init_engine(self, *args, **kwargs) -> _AsyncLLMEngine:
```
Or should we change to `Any` since the output can be a Ray remote actor?